app/src/shared/components/TextField/NumberField/index.jsx

```diff
@@ -0,0 +1,77 @@
+// @flow
+
+import React, { useCallback, useState } from 'react'
+import cx from 'classnames'
+
+import SvgIcon from '$shared/components/SvgIcon'
+
+import styles from '../textField.pcss'
+
+type Props = {
+    className?: string,
+    type?: string,
+    min?: number,
+    max?: number,
+    step?: number,
+    hideButtons?: boolean,
+}
+
+const NumberField = ({ className, hideButtons, ...props }: Props) => {
+    const [value, setValue] = useState(null)
+
+    const addValue = useCallback((step) => {
+        let valueInt = Number.parseFloat(value)
+        let stepInt = Number.parseFloat(step)
+
+        if (Number.isNaN(valueInt)) {
+            valueInt = 0
+        }
+        if (Number.isNaN(stepInt)) {
+            stepInt = 1
+        }
+
+        let newValue = valueInt + stepInt
+
+        if (props.min && newValue < props.min) {
```

Comment:
What if `props.min` is `0`?

Reply:
Also, can we have separate consts for `min` and `max`? I'd like to get rid of them repeated `props.`.